pkg/controllers/cluster/import.go

```diff
@@ -259,6 +259,18 @@ func (i *importHandler) restConfigFromKubeConfig(data []byte) (*rest.Config, err
 		return nil, err
 	}
 
+	if raw.Contexts[clusterName] != nil {
+		cluster := raw.Contexts[clusterName].Cluster
```

Comment:
can you combine the logic http://fleet.rancher.io/gitrepo-structure/ to this if? We can avoid duplicate code.